src/sentry/auth/access.py

```diff
@@ -100,45 +102,83 @@ class Access(abc.ABC):
     scopes: FrozenSet[str] = frozenset()
     permissions: FrozenSet[str] = frozenset()
 
-    member: Optional[OrganizationMember] = None
+    _member: OrganizationMember | None = None
 
     @property
-    def role(self) -> Optional[str]:
-        return self.member.role if self.member else None
+    def role(self) -> str | None:
+        return self._member.role if self._member else None
 
     @cached_property
     def _team_memberships(self) -> Mapping[Team, OrganizationMemberTeam]:
-        if self.member is None:
+        if self._member is None:
             return {}
         return {
             omt.team: omt
             for omt in OrganizationMemberTeam.objects.filter(
-                organizationmember=self.member, is_active=True, team__status=TeamStatus.VISIBLE
+                organizationmember=self._member, is_active=True, team__status=TeamStatus.VISIBLE
             ).select_related("team")
         }
 
     @cached_property
-    def teams(self) -> FrozenSet[Team]:
-        """Return the set of teams in which the user has actual membership."""
-        return frozenset(self._team_memberships.keys())
+    def team_ids_with_membership(self) -> FrozenSet[int]:
```

Comment:
Could do `membership_teams` and `accessible_teams` to get the verb,noun ordering consistent.